### PR TITLE
Add erb formatter support

### PR DESCRIPTION
**Pull request by nbelzer**
Couldn't find any guidelines to contributing so my apologies if something is not handled properly.

- Added matcher to match files ending in `.erb` to the `_Erb` language (which was unsupported before).
- Adds support for [erb-formatter](https://github.com/nebulab/erb-formatter) to format `_Erb` files.

I decided to use `_Erb` as the language-id as Github does not seem to have an official title for the language, instead identifying it as `HTML+ERB`.

### CODE REVIEW

File: format-all.el [hunk 1]
```diff
@@ -1310,6 +1319,7 @@ unofficial languages IDs are prefixed with \"_\"."
       (and (equal major-mode 'ledger-mode) "_Ledger")
       (and (equal major-mode 'nginx-mode) "_Nginx")
       (and (equal major-mode 'snakemake-mode) "_Snakemake")
+      (and (string-match-p ".erb\\'" buffer-file-name) "_Erb")
```
Comment (nbelzer) [inline format-all.el:1322]: Would love to hear suggestions on how to handle this in a better way

**Comment by nbelzer** [comment]
> We could add all of these to [language-id](https://github.com/lassik/emacs-language-id). It seems possible to tell them apart by filename extension. What Emacs modes do people use to edit ERB? I can't find an erb-mode. Does web-mode support it?

ERB is indeed supported by web-mode (which is also the major mode I am using when editing `.html.erb` files). As far as I am aware, `erb-formatter` only handles HTML files. Because of this I would suggest we only add support for `HTML+ERB`.  I'll update the matcher to be more selective based on the filename (`.html.erb` instead of just `.erb`) and update the language-id.

**Comment by nbelzer** [comment]
I've left two fixups (to make it clear what I've changed).  If you are okay with these changes I can rebase the PR to clean them up before we merge.

**Comment by lassik** [comment]
Format-all uses a separate library called language-id.el to determine what language is used in the buffer to be formatted. Only languages for which there isn't an official GitHub Linguist ID are special-cased in format-all.el as a necessary hack.

Since both HTML+ERB and JavaScript+ERB are official LInguist IDs, I added them directly to language-id.el and we don't have to add any special-case code to detect them in format-all itself.

Hence, the PR looks fine as it is, but can you remove the special-case `(and (string-match-p ".erb\\'" buffer-file-name) "HTML+ERB")`?

File: format-all.el (hunk 1 above)
Comment (nbelzer) [reply format-all.el]: This no longer applies.

File: README.md [hunk 2]
```diff
@@ -42,6 +42,7 @@ Supported languages
 * **Elixir** ([*mix format*](https://hexdocs.pm/mix/master/Mix.Tasks.Format.html))
 * **Elm** ([*elm-format*](https://github.com/avh4/elm-format))
 * **Emacs Lisp** (Emacs)
+* **Erb** ([erb-formatter](https://github.com/nebulab/erb-formatter))
```
Comment (lassik) [inline README.md:45]: Please make it `[*erb-formatter*]` like the others.

**Review by lassik — CHANGES_REQUESTED** [review]

**Comment by lassik** [comment]
Hmm. We should probably refer to the formatter as `erb-format` since that's the name of the shell command used to run it. Seems clearer to me than `erb-formatter` which is the name of the Ruby module and gem. What do you think?

**Comment by nbelzer** [comment]
Agree, also seems more in line with the other formatters.  I've made the changes

**Comment by lassik** [comment]
Perfect. Thanks a lot for sticking with it!

**Pull request merged**